hta/memory_analysis.py

```diff
@@ -0,0 +1,125 @@
+import enum
+import gzip
+import json
+import os
+
+from typing import List, Tuple
+
+import numpy as np
+import plotly
+import plotly.graph_objects as go
+
+
+colorscheme = plotly.colors.qualitative.Pastel
+
+
+class Category(enum.Enum):
+    INPUT = enum.auto()
+    TEMPORARY = enum.auto()
+    ACTIVATION = enum.auto()
+    GRADIENT = enum.auto()
+    AUTOGRAD_DETAIL = enum.auto()
+    PARAMETER = enum.auto()
+    OPTIMIZER_STATE = enum.auto()
+
+
+_CATEGORY_TO_COLORS = {
+    "Parameter": colorscheme[0],
+    "Optimizer state": colorscheme[1],
+    "Input": colorscheme[2],
+    "Temporary": colorscheme[3],
+    "Activation": colorscheme[4],
+    "Gradient": colorscheme[5],
+    "Autograd": colorscheme[6],
+    "Unknown": colorscheme[7],
+}
+
+_CATEGORY_TO_INDEX = {c: i for i, c in enumerate(_CATEGORY_TO_COLORS)}
+
+
+class MemoryAnalysis:
+    def __init__(self, path):
+        self.path = path
+
+    def _process_raw_events(self) -> Tuple[List[int], List[List[float]]]:
+        """
+        Loads the raw events and converts them into list of timestamps and list
+        of memory used by each category at each timestamp.
+
+        Returns:
+            ([timestamps], [memory_for_each_category_in_bytes])
+        """
+        file_handle = gzip.open(self.path, "r")
+        raw_timeline = json.loads(file_handle.read())
+
+        times: List[int] = []
+        sizes: List[List[float]] = []
```

Comment:
could you commone what sizes contains, list of list of byte sizes?

Reply:
see line 50